src/tir/usmp/analysis/extract_buffer_info.cc

```diff
@@ -491,6 +559,55 @@ BufferInfoAnalysis BufferInfoExtractor::operator()(const PrimFunc& main_func) {
       open_set.erase(le_event.buffer_info);
     }
   }
+
+  // All items in RO pool should have conflicts with each other in this RO pool
+  // as they will be placed in RO segment and pre-initialized
+  // Array<BufferInfo> buffer_info_arr =
+  //    ConvertToArrayOfBufferInfo(this->buffer_info_map_);
+
+  // split buffers to vars (RW) and constants (RO)
+  Array<BufferInfo> buffer_info_vars;
+  Array<BufferInfo> buffer_info_cons;
+  for (const auto& kv : this->buffer_info_map_) {
+    const auto& buf = kv.first;
+    for (const auto& pool : buf->pool_candidates) {
+      if (pool->IsInstance<ConstantPoolInfoNode>()) {
+        buffer_info_cons.push_back(buf);
+      } else {
+        buffer_info_vars.push_back(buf);
+      }
+
+      break;
+    }
+  }
+  ICHECK(buffer_info_map_.size() == buffer_info_vars.size() + buffer_info_cons.size())
+      << "missing value";
+
+  // intersect with each other, as all constants should exist at the same time
+  for (const auto& buf1 : buffer_info_cons) {
+    for (const auto& buf2 : buffer_info_cons) {
+      if (buf1->conflicts.end() ==
+          std::find(buf1->conflicts.begin(), buf1->conflicts.end(), buf2)) {
+        buf1->conflicts.push_back(buf2);
+        if (buf2->conflicts.end() ==
+            std::find(buf2->conflicts.begin(), buf2->conflicts.end(), buf1)) {
+          buf2->conflicts.push_back(buf1);
+        }
+      }
+    }
+    // remove conflicts with vars part as non-relevant anymore
```

Comment:
Is this necessary ? I do understand it is not relevant by the virtue of now we will only place constants in a different pool but therefore liveness across pools does not matter. In the same time, will it break if we have every buffer to be a conflicting with constant nodes ?

Reply:
this is needed